tests/Unit/inc/common/doAdminPostRocketPurgeCache.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace WP_Rocket\Tests\Unit\Inc\Common;
+
+use Brain\Monkey\Actions;
+use Brain\Monkey\Functions;
+use WPMedia\PHPUnit\Unit\TestCase;
+
+/**
+ * @covers ::do_admin_post_rocket_purge_cache
+ * @group Common
+ */
+class Test_DoAdminPostRocketPurgeCache extends TestCase {
+
+	protected function setUp() {
+		parent::setUp();
+
+		Functions\when( 'sanitize_key' )->alias(
+			function( $key ) {
+				$key = strtolower( $key );
+				return preg_replace( '/[^a-z0-9_\-]/', '', $key );
+			}
+		);
+
+		Functions\when( 'absint' )->alias(
+			function( $maybeint ) {
+				return abs( intval( $maybeint ) );
+			}
+		);
+
+		Functions\when( 'wp_unslash' )->alias(
+			function( $value ) {
+				return stripslashes( $value );
+			}
+		);
+
+		Functions\when( 'sanitize_title' )->alias(
+			function( $value ) {
+				return strtolower( $value );
+			}
+		);
+
+		Functions\when( 'get_option' )->justReturn( '' );
+		Functions\when( 'wp_verify_nonce' )->justReturn( true );
+		Functions\when( 'current_user_can' )->justReturn( true );
+		Functions\when( 'set_transient' )->justReturn( null );
+		Functions\when( 'wp_get_referer' )->justReturn( 'http://example.org' );
+		Functions\when( 'esc_url_raw' )->returnArg();
+		Functions\when( 'wp_safe_redirect' )->justReturn( null );
+		Functions\when( 'wp_nonce_ays' )->justReturn( null );
+
+		require_once WP_ROCKET_PLUGIN_ROOT . 'inc/common/purge.php';
+
+		Functions\when( 'rocket_clean_post' )->justReturn( null );
```

Comment:
Since the function is defined in the same file, you don't need to do that. Instead you can create stubs for the functions inside  `rocket_clean_post`

Reply:
It may be defined in the same file, but as a public function, it has (or should have) its own tests. Not stubbing it here would mean we also test it here, which we don't want. That's why I believe it must be stubbed.

Reply:
It does need to be stubbed _when_ it's used in the `do_admin_post_rocket_purge_cache()` callback function.

I'm looking at the unit test environment now to find the root cause.